microsoft-edge/web-platform/ad-selection-api.md

```diff
@@ -0,0 +1,225 @@
+
+
+<!-- ------------------------------ -->
+#### Buyer's Service
+
+* **BuyerFrontEnd Service:** The front-end service provides a gRPC endpoint `/GetBids`, which receives requests from **SellerFrontEnd** to initiate the bidding flow.
+
+* **Bidding Service:** This service provides an endpoint `/GenerateBids`, which receives requests from the **BuyerFrontEnd** service to handle the bidding and generate a bid.  The service generates a bid, chooses the winner, and selects the banner to be rendered.
+
+* **Buyer's Key/Value Service:** A buyer's Key/Value service receives requests from the **BuyerFrontEnd** service and returns real-time buyer data required for bidding, corresponding to lookup keys from Interest Groups.  This request happens once per workflow.  The Key/Value service will be run in BYOS mode.
+
+* **Bidding Selection & Key/Value Service:** This service receives requests from the Bidding Service to select and return additional ad banners (candidates) that can participate in bidding.  The service can also return additional signals needed for bidding.  The Bidding Service may send multiple requests to the Selection & Key/Value service or may choose not to send any, as the service is optional.  The Selection & Key/Value service must be deployed in a TEE.
+
+* **K-Anon Service:** This service collects k-anonymity counters and checks that the winning ad banner passes the k-anonymity check.
+
+Buyers and sellers need to provide their own custom code as User-Defined Functions (UDFs) that will run in private containers within the deployed services.  These User-Defined Functions can execute custom business logic.
+
+
+<!-- ------------------------------ -->
+#### Seller's User Define Function
+
+* `scoreAd()` for **Auction Service:** This function should generate a score for each buyer's bid or reject it.  The score will be used in the **SellerFrontEnd** to choose a winner from among all buyers.
+
+* `reportResult()` for **Event-Level Reporting:** This function will run in the **SellerFrontEnd** after the final winner has been chosen.  The goal of `reportResult()` is to notify the seller about the winning bidder and provide the bid value.
+
+
+<!-- ------------------------------ -->
+#### Buyer's User Define Function
+
+* `generateBids()` for **Bidding Service:**  This function will generate a bid and choose the banner that will be shown.
+
+* `reportWin()` for **Event-Level Reporting:**  This function will run in the **SellerFrontEnd** if the buyer's bid wins the auction.  The goal of `reportWin()` is to notify the buyer that they have won the auction and generate notification URLs that will be triggered later, such as during banner rendering or other client events.
+
+* `getValues()` for **Bidding's Selection & Key/Value Service:**  This function will run in the Bidding Selection & Key/Value service if the buyer uses this service.  It allows for custom code execution for data lookup and selection.
```

Comment:
is & technically correct and part of a UI string here?
per corp style guide, avoid &  unless it's in UI
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/a/ampersand

maybe:
run in the Foo service and in the Bar service

Global in .md file
@captainbrosset